replace function get_avatar_url

Grexy opened the pull request:
As of WordPress 4.2, there is now has a built in get_avatar_url function.  Removing custom function and rewritten to use WordPress function.

retlehs commented on lib/discourse.php:
```diff
@@ -205,7 +207,7 @@ function sso_parse_request( $wp )
           'about_me' => $current_user->description,
           'external_id' => $current_user->ID,
-          'avatar_url' => self::get_avatar_url($current_user->ID)
+          'avatar_url' => $avatar
```
can you just make this `'avatar_url' => get_avatar_url(get_current_user_id())` and get rid of the new variables?

nathanielks replied:
dat perf!

retlehs commented:
@Grexy thanks! do you mind squashing this into one commit? if you haven't done it before here's a guide: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

Grexy commented:
sorry i am a bit of a noob to this.  so i rebased and squashed the commits into a single one here.  https://github.com/Grexy/wp-discourse/commit/1ca0a85e6e1d321169aee8e8fa7cdaf380c2946e.  Cant figure out how to sync this with my pull request.
